### PR TITLE
Bundle Linux Libertine Fonts

### LINKED ISSUE #1292: Bundle our own fonts
**Is your feature request related to a problem? Please describe.**
Being a cross-platform application we rely on the available set of fonts across many OS ecosystems. To better control the look at feel of the text in fc21, I think we should look at bundling our own TTF fonts and reference those in the options dialog.

**Describe the solution you'd like**
We need to find an open source licensed font set that supports a wide array of languages and character sets. The Liberation Fonts first written by Red Hat -- https://github.com/liberationfonts/liberation-fonts -- is a potential candidate. They use an open source license and support a wide array of languages. Another option is the GNU FreeFont set -- https://www.gnu.org/software/freefont/ -- that is licensed GPL same as fc21 and also supports a huge array of languages.

This enhancement will require a change to the packages as well to ensure the font files are included.

**Describe alternatives you've considered**
None

**Additional context**
This issue came up while working on a font bug fix in PR #1281

### LINKED ISSUE #1341: Move macos/info.plist to dist/info.plist
**Is your feature request related to a problem? Please describe.**
We use `macos/info.plist` as part of the macOS packaging process. This file really should be in `dist/` along with the other distribution files.

**Describe the solution you'd like**
Move the file, remove the `macos` directory and update the `build.yaml` file to capture the change.

**Describe alternatives you've considered**
None

**Additional context**
I found this as part of working on #916

### PR DESCRIPTION
**Pull request by jwrober**
Instead of adding the files directly to the repo, we use a `cmake` feature to go fetch the files at install and package time. We also set the dependency for the `.deb` package. Will need @pranavtifr to update the AUR package to match.

Closes #1292 
Closes #1341

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -342,5 +342,31 @@ if(UNIX AND NOT APPLE)
   endif(FREECIV_ENABLE_RULEDIT)
 endif(UNIX AND NOT APPLE)
 
+# We grab the Libertine Font package online for non-Unix (e.g. Win32, MSYS2, and macOS)
+#  For Linux, we add a dependency to the package.
+if(NOT UNIX)
```
Comment (lmoureaux) [inline CMakeLists.txt:347]: `UNIX` is true on Mac as well. I suggest making this a configure-time option.
Also, this should only be downloaded when we build the client (thinking of the LT server where we have a restrictive proxy).

File: CMakeLists.txt [hunk 2]
```diff
@@ -342,5 +342,31 @@ if(UNIX AND NOT APPLE)
...
+  message(STATUS "Downloading Linux Libertine Font Package")
+
+  include(ExternalProject)
+  ExternalProject_Add(Libertine
+    PREFIX ${CMAKE_BINARY_DIR}
+    LOG_DIR ${CMAKE_BINARY_DIR}
+    DOWNLOAD_NO_PROGRESS TRUE
+    DOWNLOAD_EXTRACT_TIMESTAMP TRUE
+    TIMEOUT 30
```
Comment (lmoureaux) [inline CMakeLists.txt:356]: 30s for 3MB assumes that the user has a bandwidth of 100kB/s. Let's increase this a bit (5-10 minutes to be more on the safe side?).

File: CMakeLists.txt (hunk 1 above)
Comment (lmoureaux) [inline CMakeLists.txt:347]: Maybe move to a separate file in `cmake/` to avoid cluttering the main CMakeLists too much.

File: CMakeLists.txt [hunk 3]
```diff
@@ -342,5 +342,31 @@ if(UNIX AND NOT APPLE)
...
+    CONFIGURE_COMMAND ""
+    BUILD_COMMAND ""
+    INSTALL_COMMAND ""
+    URL https://sourceforge.net/projects/linuxlibertine/files/linuxlibertine/5.3.0/LinLibertineOTF_5.3.0_2012_07_02.tgz
```
Comment (lmoureaux) [inline CMakeLists.txt:360]: Can we add `URL_HASH` to ensure integrity?

File: CMakeLists.txt [hunk 4]
```diff
@@ -342,5 +342,31 @@ if(UNIX AND NOT APPLE)
...
+  )
+
+  install(
+    DIRECTORY ${CMAKE_BINARY_DIR}/src/Libertine
+    DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/fonts
```
Comment (lmoureaux) [inline CMakeLists.txt:365]: This installs to `/usr/share/fonts` on Linux and could overwrite some system files. Better put them in our private data dir, `SHARE/freeciv21/fonts`.

**Review by lmoureaux — CHANGES_REQUESTED** [review]

**Comment by pranavtifr** [comment]
For future reference, this needs to be added as a dependency
https://archlinux.org/packages/community/any/ttf-linux-libertine/

File: CMakeLists.txt (hunk 1 above)
Comment (jwrober) [reply CMakeLists.txt:347]: That is the intention (to get picked up on all but Linux/Unix). I'll add another condition so it only runs if you are building the client.

File: CMakeLists.txt (hunk 2 above)
Comment (jwrober) [reply CMakeLists.txt:356]: Yea looking at the documentation, I need to bump this up some to handle slower connections.

File: CMakeLists.txt (hunk 1 above)
Comment (jwrober) [reply CMakeLists.txt:347]: Maybe move all the install stuff at the bottom to another file?

File: CMakeLists.txt (hunk 3 above)
Comment (jwrober) [reply CMakeLists.txt:360]: The project does not offer a hash file. I would have included if one was available.

File: CMakeLists.txt (hunk 1 above)
Comment (lmoureaux) [reply CMakeLists.txt:347]: What if I'm on Slitaz and I don't have the `ttf-libertine` package though? I'm for downloading it by default and allowing distributors to use the system version

**Comment by jwrober** [comment]
> For future reference, this needs to be added as a dependency https://archlinux.org/packages/community/any/ttf-linux-libertine/

Is there an otf version of the package? The Debian one is otf instead of ttf.

File: CMakeLists.txt (hunk 1 above)
Comment (jwrober) [reply CMakeLists.txt:347]: LoL, now that is an edge case since we don't technically support that distro. I'll change it.

**Comment by lmoureaux** [comment]
> > For future reference, this needs to be added as a dependency https://archlinux.org/packages/community/any/ttf-linux-libertine/
> 
> Is there an otf version of the package? The Debian one is otf instead of ttf.

The Arch package has both ttf and otf, click on "View the file list for ttf-linux-libertine" to confirm. ttf- is just a prefix for font packages in Arch (though otf- does exist).

**Review by lmoureaux — APPROVED** [review]

**Pull request merged**